File: Assets/_Scripts/Managers/GameManager.cs
using Assets._Scripts.States;
using RageCure.EventUtils;
using RageCure.ReduxUtils;
using UnityEngine;

namespace Assets._Scripts.Managers
{
    public class GameManager : EventAggregator
    {
        //public Store<GameModel, GameAction> Store;

        public void Start()
        {
            AkSoundEngine.PostEvent("StartGame", gameObject);
        }

        #region Singleton
        public static GameManager Instance;

        public void Awake()
        {
            Instance = this;
        }

        #endregion
    }
}

File: Assembly-FSharp/Assembly-FSharp/Managers/GameManager.fs
namespace Styx.Managers

open UnityEngine
open Styx.Commons.Wwise

type GameManager () =
    inherit MonoBehaviour ()

    //*********************** SINGLETON ***************************************
    static let mutable instance: GameManager = Unchecked.defaultof<GameManager>
    static member Instance
        with get() = instance
    //*************************************************************************

    member this.Awake () = 
        instance <- this

    member this.Start() =
        SoundEngine.Instance.PostEvent("StartGame", this.gameObject)